### PR TITLE
make medusa compatible with imdsv2 on AWS

### PR DESCRIPTION
**Pull request by optplx**
For security reasons, we will enforce imdsv2 on our instances (see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html)

This is compatible with both v1 and v2 without having to worry about it

### CODE REVIEW

**Comment by optplx** [comment]
hi @adejanovski do you think this is acceptable?

File: medusa/storage/s3_storage.py [hunk 1]
```diff
@@ -82,7 +89,7 @@ def connect_storage(self):
                 logging.debug('Reading AWS credentials from IAM Role: %s', aws_instance_profile.text)
                 url = "http://169.254.169.254/latest/meta-data/iam/security-credentials/" + aws_instance_profile.text
                 try:
-                    auth_data = requests.get(url).json()
+                    auth_data = requests.get(url, headers=self.imds_headers).json()
```
Comment (adejanovski) [inline medusa/storage/s3_storage.py:92]: Do you know if this has an impact with IMDSv1 usage of IAM roles (if that's even a thing)?
ie: is this change transparent to any existing user relying on IAM roles?

Comment (optplx) [reply medusa/storage/s3_storage.py:92]: yes as per my tests: if imdsv2 is available it will use it, if not it uses imdsv1 and it is transparent

Comment (optplx) [reply medusa/storage/s3_storage.py:92]: e.g with `headers=None`: 
```
requests.get("http://169.254.169.254/latest/meta-data/iam/security-credentials/xxxxx", headers=None).text
'{\n  "Code" : "Success",\n  "LastUpdated" : "2021-09-07T08:02:16Z",\n  "Type" : "AWS-HMAC",\n  "AccessKeyId" : "xxxx",\n  "SecretAccessKey" : "xxxxxxx",\n  "Token" : "xxxx==",\n  "Expiration" : "2021-09-07T14:20:41Z"\n}'
```

Comment (adejanovski) [reply medusa/storage/s3_storage.py:92]: Sounds good, thanks!

**Review by adejanovski — APPROVED** [review]

**Comment by adejanovski** [comment]
I should've seen there were 6 commits before merging... My bad 😓

**Comment by optplx** [comment]
> I should've seen there were 6 commits before merging... My bad 😓

yes I tested the "rebase" feature from the console quite a lot, sorry :p

**Pull request merged**